src/Symfony/Bridge/Twig/Translation/TwigExtractor.php

```diff
@@ -52,16 +52,22 @@ public function __construct(\Twig_Environment $twig)
     /**
      * {@inheritdoc}
      */
-    public function extract($directory, MessageCatalogue $catalogue)
+    public function extract($resource, MessageCatalogue $catalogue)
     {
-        // load any existing translation files
-        $finder = new Finder();
-        $files = $finder->files()->name('*.twig')->sortByName()->in($directory);
+        if (is_array($resource)) {
+            $files = array_filter($resource, function ($filename) {
```

Comment:
1. Anyway we need to be sure that file exist, the Finder is  expensive too
2. ok
3. Yes, it keep the code from below the same.